src/main/java/github/scarsz/discordsrv/modules/alerts/AlertListener.java

```diff
@@ -0,0 +1,281 @@
+import github.scarsz.discordsrv.objects.MessageFormat;
+import github.scarsz.discordsrv.util.*;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.entities.TextChannel;
+import org.apache.commons.lang3.StringUtils;
+import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Event;
+import org.bukkit.event.*;
+import org.bukkit.event.player.PlayerCommandPreprocessEvent;
+import org.bukkit.event.player.PlayerEvent;
+import org.bukkit.plugin.RegisteredListener;
+
+import java.awt.*;
+import java.lang.reflect.Field;
+import java.util.List;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class AlertListener implements Listener {
+
+    static {
+        List<HandlerList> globalHandlerListLocal = null;
+        try {
+            Field field = HandlerList.class.getDeclaredField("allLists");
+            field.setAccessible(true);
+            globalHandlerListLocal = new ArrayList<HandlerList>((List<HandlerList>) field.get(null)) {
+                @Override
+                public boolean add(HandlerList list) {
+                    boolean added = super.add(list);
+                    if (Arrays.stream(list.getRegisteredListeners()).noneMatch(listener::equals)) list.register(listener);
+                    return added;
+                }
+            };
+            field.set(null, globalHandlerListLocal);
+        } catch(NoSuchFieldException | IllegalAccessException e) {
+            e.printStackTrace();
+        }
+        HANDLERS = globalHandlerListLocal;
+    }
+
+    private static final List<HandlerList> HANDLERS;
+    private static final Set<String> SEEN_EVENTS = new HashSet<>();
+    private static RegisteredListener listener;
+
+    public void register() {
+        listener = new RegisteredListener(
+                new Listener() {},
+                (listener, event) -> onEvent(event),
+                EventPriority.MONITOR,
+                DiscordSRV.getPlugin(),
+                false
+        );
+
+        Set<String> events = DiscordSRV.config().dget("Alerts").children()
+                .map(dynamic -> dynamic.get("Trigger"))
+                .filter(Weak::isPresent)
+                .map(dynamic -> dynamic.convert().intoString())
+                .filter(s -> !s.startsWith("/"))
+                .collect(Collectors.toSet());
+
+        DiscordSRV.debug("Registered alerts for " + String.join(", ", events));
+
+        long count = DiscordSRV.config().dget("Alerts").children().count();
+        if (count > 0) DiscordSRV.info(count + " alert" + (count > 1 ? "s" : "") + " registered");
+    }
+
+    public void unregister() {
+        for (HandlerList handlerList : HandlerList.getHandlerLists()) {
+            handlerList.unregister(listener);
+        }
+    }
+
+    public <E extends Event> void onEvent(E event) {
+        if (event instanceof PlayerCommandPreprocessEvent) {
```

Comment:
Maybe allow console/rcon too, `ServerCommandEvent` / `RemoteServerCommandEvent`; though I see this isn't fully finished yet

Reply:
[`bf45544`](https://github.com/DiscordSRV/DiscordSRV/pull/759/commits/bf45544b999747c71f1ac15ecb0f4c0bed9cd63c)